### PR TITLE
Fix regression tests for concurrent CI builds

### PR DESCRIPTION
**Pull request by rickducott**

### CODE REVIEW

**Comment by rickducott** [comment]
Ran a few parallel builds, several succeeded and one failed with:
```
{"level":"fatal","ts":1552695371.5344677,"logger":"gateway","caller":"setuputils/main_setup.go:66","msg":"error in setup: creating kube resource gateway: gateways.gateway.solo.io is forbidden: User \"system:serviceaccount:test-12c36bac-6807-4175-b6c0-504ee6ef1cef:default\" cannot create gateways.gateway.solo.io in the namespace \"test-12c36bac-6807-4175-b6c0-504ee6ef1cef\"","stacktrace":"github.com/solo-io/gloo/pkg/utils/setuputils.Main\n\t/workspace/gopath/src/github.com/solo-io/gloo/pkg/utils/setuputils/main_setup.go:66\ngithub.com/solo-io/gloo/projects/gateway/pkg/setup.Main\n\t/workspace/gopath/src/github.com/solo-io/gloo/projects/gateway/pkg/setup/setup.go:9\nmain.main\n\t/workspace/gopath/src/github.com/solo-io/gloo/projects/gateway/cmd/main.go:19\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:200"}
```

**Comment by rickducott** [comment]
Making progress... 4/5 concurrent builds passed:
image


Now investigating this knative error:
```
Step #14 - "regression-tests": [kubectl apply -f -] failed: Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling admission webhook "webhook.serving.knative.dev": Post https://webhook.knative-serving.svc:443/?timeout=30s: no endpoints available for service "webhook"
```

File: install/helm/gloo/templates/2-namespace-clusterrolebinding.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 kind: ClusterRoleBinding
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  name: gloo-role-binding
+  name: gloo-role-binding-{{ .Release.Namespace }}
```
Comment (yuval-k) [inline install/helm/gloo/templates/2-namespace-clusterrolebinding.yaml]: Namespaced ClusterRoleBinding = Role?
or do you need access to resources outside the namespaces?

File: changelog/v0.11.2/fix-regression-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    description: Change name of cluster role binding created during Gloo installation to include namespace, so that writing it doesn't remove the cluster role binding for other Gloo installations.
+    issueLink: https://github.com/solo-io/gloo/issues/540
+  - type: NON_USER_FACING
+    description: Lock around KNative tests
+    issueLink: https://github.com/solo-io/gloo/issues/543
+  - type: NON_USER_FACING
+    description: Regression tests now run against the kube2e cluster.
+    issueLink: https://github.com/solo-io/gloo/issues/541
+  - type: FIX
+    description: Give the gloo cluster role the union of permissions necessary for gateway, ingress, and knative modes, to support installing multiple modes of Gloo on a single cluster.
```
Comment (yuval-k) [inline changelog/v0.11.2/fix-regression-tests.yaml]: as an alternative - maybe separate use roles ? 
that way when installing other modes the new roles will be installed.

**Comment by rickducott** [comment]
Strange message: 

```
{"level":"error","ts":1552701929.444769,"logger":"uds.v1.event_loop.uds.v1.event_loop.syncer","caller":"syncer/discovery_syncer.go:51","msg":"error in UDS: reconciling 39 desired upstreams: reconciling resource test-eefbfcc2-fe54-4507-ae8e-327063bc297d-clusteringress-proxy-: validation error: a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')","stacktrace":"github.com/solo-io/gloo/projects/discovery/pkg/uds/syncer.(*syncer).Sync.func1\n\t/workspace/gopath/src/github.com/solo-io/gloo/projects/discovery/pkg/uds/syncer/discovery_syncer.go:51"}
```

**Comment by rickducott** [comment]
The latest change seemed to make the locking only last aroune 16 seconds (2 seconds * 8 attempts), expected backoff

File: changelog/v0.11.2/fix-regression-tests.yaml (hunk 2 above)
Comment (rickducott) [reply changelog/v0.11.2/fix-regression-tests.yaml]: Done in #553

**Comment by rickducott** [comment]
This was useful for testing but closing out now... I have ported over several of the fixes, and will be pushing the gateway fix in an upcoming PR.

**Pull request closed without merging**